saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationFilter.java

```diff
@@ -95,6 +101,7 @@ protected boolean requiresAuthentication(HttpServletRequest request, HttpServlet
 	public Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response)
 			throws AuthenticationException {
 		Authentication authentication = this.authenticationConverter.convert(request);
+		this.authenticationRequestRepository.removeAuthenticationRequest(request, response);
```

Comment:
It seems that if `authentication == null` this might be an invalid request. Should the authnrequest be removed after it's clear that `authentication != null`?

Reply:
Yes, it should.